Library/Homebrew/completions/zsh.erb

```diff
@@ -49,6 +49,8 @@ __brew_completion_caching_policy() {
 }
 
 __brew_formulae() {
+  [[ -prefix '-' ]] && return 0
```

Comment:
```suggestion
  [[ -prefix '-' ]] && return 0
  [[ -after '--cask' ]] && return 0
```

I've only briefly skimmed the docs and I haven't tested this, but will this work to ensure that formulae are not completed when `--cask` is passed?

And likewise, `[[ -after '--formula' ]] && return 0` can be added to the `__brew_*casks` functions as well

Reply:
I looked into this briefly but I stumbled into some problems where if you wanted to specify multiple formulae to something like `brew install` it would stop doing completions. I'll dig into this later.